src/socket.cpp

```diff
@@ -1685,7 +1685,11 @@ void SIPpSocket::sipp_sctp_peer_params()
         memset(&peerparam, 0, sizeof(peerparam));
 
         sockaddr* addresses;
+#ifdef __SUNOS
+        int addresscount = sctp_getpaddrs(ss_fd, 0, (void **) &addresses);
+#else
         int addresscount = sctp_getpaddrs(ss_fd, 0, &addresses);
```

Comment:
How about casting to `(struct sockaddr**)` instead. Then you can drop the `#ifdef`.